sql/mysqld.cc

```diff
@@ -1489,6 +1489,8 @@ static void openssl_lock(int, openssl_lock_t *, const char *, int);
 char *des_key_file;
 #ifndef EMBEDDED_LIBRARY
 struct st_VioSSLFd *ssl_acceptor_fd;
+static char upgrade_info_file[FN_REFLEN]= {0};
```

Comment:
Does this need to be a global?

Reply:
The idea behind having this as a global variable is to just build the path once and use it every time we check. On the revision you checked the initial periodic check was removed. I've reverted that on my latest update (including the other requested changes) as I think we should include the periodic check, you can find more context here: https://github.com/MariaDB/server/pull/2464#discussion_r1090082443.